File: AlgorithmDesign/intmath/euclidean.c
#include <stdio.h>
#include <stdlib.h>

int gcd(int a, int b) {
  int c;
  
  do { /* */ printf("a = %d, b = %d, ", a, b);
    c = a % b; /* */ printf("a mod b = %d;\n", c);
    a = b;
    b = c;
  } while ( c != 0 );
  return a;
}

int main(int argc, char * argv[]) {
  int a, b, divisor;

  if (argc != 3) 
    return 1;
  a = atoi(argv[1]);
  b = atoi(argv[2]);
  divisor = gcd(a, b);

  printf("gcd of %d and %d is: %d.\n", a, b, divisor);

  return 0;
}

File: AlgorithmDesign/intmath/Euclidean.java
import java.lang.Integer;

class Euclidean {

    static long gcd(long a, long b) {
	long c;
	
	do { // System.err.print("a = " + a + ", b = " + b);
	    c = a % b; // System.err.println(", a mod b = " + c + ";");
	    a = b;
	    b = c;
	} while ( c != 0 );
	return a;
    }

    static public void main(String argv[]) {
	long a, b, divisor;

	if (argv.length != 2) {
	    System.err.println("Arguments are too few or many.");
	    return;
	}
	a = Integer.parseInt(argv[0]);
	b = Integer.parseInt(argv[1]);

	divisor = gcd(a, b);

	System.out.println("gcd of " + a + " and " + b + " is: " 
			   + divisor + ".");

	return;
    }
};
